scripts/apache/get_fasta_sequence

```diff
@@ -29,49 +29,27 @@ sub get_fasta_sequence {
     my ($server) = @_;
 
     my $id = $server->require_argument('id');
+    my $seq_target = $server->require_argument('sequence_target');
 
-    my @raw_id_array = split (',', $id);
-    $id = "";
-    foreach my $current_id (@raw_id_array) {
-        if (index($current_id, '-') != -1) {
-          $current_id =~ s/\.\d+//;
-        }
-        $id = $id . ',' . $current_id;
-    }
-
-    my @queries;
-    my $query = 'http://www.ebi.ac.uk/Tools/dbfetch/dbfetch?format=fasta&style=raw&db=uniprotkb&id=';
-    push @queries, $query;
-    $query = 'http://www.ebi.ac.uk/Tools/dbfetch/dbfetch?format=fasta&style=raw&db=refseqn&id=';
-    push @queries, $query;
-    $query = 'http://www.ebi.ac.uk/Tools/dbfetch/dbfetch?db=ena_sequence&format=fasta&style=raw&Retrieve=Retrieve&id=';
-
-    push @queries, $query;
+    # my @queries;
+    # my $query = 'http://www.ebi.ac.uk/Tools/dbfetch/dbfetch?format=fasta&style=raw&db=uniprotkb&id=';
+    # push @queries, $query;
+    # $query = 'http://www.ebi.ac.uk/Tools/dbfetch/dbfetch?format=fasta&style=raw&db=refseqn&id=';
+    # push @queries, $query;
+    # $query = 'http://www.ebi.ac.uk/Tools/dbfetch/dbfetch?db=ena_sequence&format=fasta&style=raw&Retrieve=Retrieve&id=';
+    # push @queries, $query;
 
     my $response;
-    my $concat_response = '';
-    my $i = 0;
-    while ($i < 3 || $id != '') {
-        my @id_array = split (',', $id);
-        $id = '';
-        foreach my $current_id (@id_array) {
-            if ((index $concat_response, $current_id) == -1) {
-                warn "ID " . $current_id . "not found ";
-                 $id = $id . ',' . $current_id;
-            }
-        }
-
-        my $cut_id = $id;
 
-        if ($i == 0) {
-            $cut_id =~ s/\.\d+//;
-        }
-
-        $response = fetch_ids_from_external_db($queries[$i] . $cut_id);
-        $concat_response = $concat_response . $response;
-        $i++;
+    if($seq_target eq 'dna') {
+        $response = fetch_ids_from_external_db('http://www.ebi.ac.uk/Tools/dbfetch/dbfetch?format=fasta&style=raw&db=refseqn&id=' . $id);
+    }
+    if($seq_target eq 'protein') {
+        $id =~ s/\.\d+//;
+        $response = fetch_ids_from_external_db('http://www.ebi.ac.uk/Tools/dbfetch/dbfetch?format=fasta&style=raw&db=uniprotkb&id=' . $id);
```

Comment:
It was three sources before changes, three should be after changes.  Two db's for dna and one for protein.

Reply:
I remember someone mentioning we should fetch DNA sequence from refseq only and protein from uniprot.

Reply:
It should be explicitly confirmed by some of senior annotators. Removing one of db's is a serious change